src/sentry/release_health/release_monitor/metrics.py

```diff
@@ -0,0 +1,152 @@
+                )
+                data = raw_snql_query(
+                    query, referrer="release_monitor.fetch_projects_with_recent_sessions"
+                )["data"]
+                count = len(data)
+                more_results = count > self.CHUNK_SIZE
+                offset += self.CHUNK_SIZE
+
+                if more_results:
+                    data = data[:-1]
+
+                for row in data:
+                    aggregated_projects[row["org_id"]].append(row["project_id"])
+
+                if not more_results:
+                    break
+
+            else:
+                logger.error(
+                    "release_monitor.fetch_projects_with_recent_sessions.loop_timeout",
+                    extra={"offset": offset},
+                )
+
+        return aggregated_projects
+
+    def fetch_project_release_health_totals(
+        self, org_id: int, project_ids: Sequence[int]
+    ) -> Totals:
+        start_time = time.time()
+        offset = 0
+        totals: Totals = defaultdict(dict)
+        with metrics.timer("release_monitor.fetch_project_release_health_totals.loop"):
+            while (time.time() - start_time) < self.MAX_SECONDS:
+                with metrics.timer("release_monitor.fetch_project_release_health_totals.query"):
+                    query = (
+                        Query(
+                            dataset=Dataset.Metrics.value,
+                            match=Entity(EntityKey.MetricsCounters.value),
+                            select=[
+                                # TODO: What will this be? Sum of value?
+                                Column("sessions"),
+                            ],
+                            groupby=[
+                                Column("org_id"),
+                                Column("project_id"),
+                                Column("release"),
+                                Column("environment"),
+                            ],
+                            where=[
+                                Condition(
+                                    Column("timestamp"),
+                                    Op.GTE,
+                                    datetime.utcnow() - timedelta(hours=6),
+                                ),
+                                Condition(Column("timestamp"), Op.LT, datetime.utcnow()),
+                                Condition(Column("org_id"), Op.EQ, org_id),
+                                Condition(Column("project_id"), Op.IN, project_ids),
+                                Condition(
+                                    Column("metric_id"),
+                                    Op.EQ,
+                                    SESSION_METRIC_NAMES["c:sessions/session@none"],
+                                ),
+                            ],
+                            granularity=Granularity(21600),
+                            orderby=[
+                                OrderBy(Column("org_id"), Direction.ASC),
```

Comment:
Probably also enough to orderBy `project_id` in this case

Reply:
Actually realised I need to order by `project_id, release, env` just to keep the sort stable. That's probably a bug in the old system, although likely doesn't hurt us much